hedera-node/src/main/java/com/hedera/services/context/ImmutableStateChildren.java

```diff
@@ -0,0 +1,167 @@
+ */
+public class ImmutableStateChildren implements StateChildren {
+	private WeakReference<MerkleMap<EntityNum, MerkleAccount>> accounts;
+	private WeakReference<MerkleMap<EntityNum, MerkleTopic>> topics;
+	private WeakReference<MerkleMap<EntityNum, MerkleToken>> tokens;
+	private WeakReference<MerkleMap<EntityNumPair, MerkleUniqueToken>> uniqueTokens;
+	private WeakReference<MerkleMap<EntityNum, MerkleSchedule>> schedules;
+	private WeakReference<VirtualMap<VirtualBlobKey, VirtualBlobValue>> storage;
+	private WeakReference<VirtualMap<ContractKey, ContractValue>> contractStorage;
+	private WeakReference<MerkleMap<EntityNumPair, MerkleTokenRelStatus>> tokenAssociations;
+	private WeakReference<MerkleNetworkContext> networkCtx;
+	private WeakReference<AddressBook> addressBook;
+	private WeakReference<MerkleSpecialFiles> specialFiles;
+	private WeakReference<RecordsRunningHashLeaf> runningHashLeaf;
+	private WeakReference<Map<ByteString, EntityNum>> aliases;
+	private Instant signedAt = Instant.EPOCH;
+	private ServicesState state;
+
+	public ImmutableStateChildren(ServicesState state) {
+		this.state = state;
+	}
+
+	@Override
+	public Instant signedAt() {
+		return signedAt;
+	}
+
+	@Override
+	public MerkleMap<EntityNum, MerkleAccount> accounts() {
+		final var refAccounts = accounts.get();
+		Objects.requireNonNull(refAccounts);
+		return refAccounts;
+	}
+
+	@Override
+	public MerkleMap<EntityNum, MerkleTopic> topics() {
+		return Objects.requireNonNull(topics.get());
+	}
+
+	@Override
+	public MerkleMap<EntityNum, MerkleToken> tokens() {
+		return Objects.requireNonNull(tokens.get());
+	}
+
+	@Override
+	public VirtualMap<VirtualBlobKey, VirtualBlobValue> storage() {
+		return Objects.requireNonNull(storage.get());
+	}
+
+	@Override
+	public VirtualMap<ContractKey, ContractValue> contractStorage() {
+		return Objects.requireNonNull(contractStorage.get());
+	}
+
+	@Override
+	public MerkleMap<EntityNum, MerkleSchedule> schedules() {
+		return Objects.requireNonNull(schedules.get());
+	}
+
+	@Override
+	public MerkleMap<EntityNumPair, MerkleTokenRelStatus> tokenAssociations() {
+		return Objects.requireNonNull(tokenAssociations.get());
+	}
+
+	@Override
+	public MerkleNetworkContext networkCtx() {
+		return Objects.requireNonNull(networkCtx.get());
+	}
+
+	@Override
+	public AddressBook addressBook() {
+		return Objects.requireNonNull(addressBook.get());
+	}
+
+	@Override
+	public MerkleSpecialFiles specialFiles() {
+		return Objects.requireNonNull(specialFiles.get());
+	}
+
+	@Override
+	public MerkleMap<EntityNumPair, MerkleUniqueToken> uniqueTokens() {
+		return Objects.requireNonNull(uniqueTokens.get());
+	}
+
+	@Override
+	public RecordsRunningHashLeaf runningHashLeaf() {
+		return Objects.requireNonNull(runningHashLeaf.get());
+	}
+
+	@Override
+	public Map<ByteString, EntityNum> aliases() {
+		return Objects.requireNonNull(aliases.get());
+	}
+
+	public void updatePrimitiveChildren(final Instant signingTime) {
```

Comment:
For the `Immutable` version, we can make all fields `final` and remove this method. We can set the `signingTime` in the constructor like,
```
signedAt = signedState.getTimeOfLastHandledTxn();
```

Reply:
Set all the variables in constructor and removed `updatePrimitiveChildren` method